### ISSUE TITLE
1.1.0: export --dev does not include sub-dependencies

### ISSUE THREAD

**Issue by cjolowicz** [python-poetry/poetry#3074]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 10.14
- **Poetry version**: 1.1.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: (not needed, reproducible with a default project generated by poetry new)

## Issue

Since Poetry 1.1.0, the export command no longer includes sub-dependencies for development dependencies in the generated `requirements.txt`.

To reproduce:

```sh
$ poetry new example
$ cd example
$ poetry export -f requirements.txt --dev
pytest==5.4.3 \
    --hash=sha256:5c0db86b698e8f170ba4582a492248919255fcd4c79b1ee64ace34301fb589a1 \
    --hash=sha256:7979331bfcba207414f5e1263b5a0f8f521d0f457318836a7355531ed1a4c7d8
```

For comparison, the output with Poetry 1.0.10 is the following:

```
atomicwrites==1.4.0; sys_platform == "win32" \
    --hash=sha256:6d1784dea7c0c8d4a5172b6c620f40b6e4cbfdf96d783691f2e1302a7b88e197 \
    --hash=sha256:ae70396ad1a434f9c7046fd2dd196fc04b12f9e91ffb859164193be8b6168a7a
attrs==20.2.0 \
    --hash=sha256:fce7fc47dfc976152e82d53ff92fa0407700c21acd20886a13777a0d20e655dc \
    --hash=sha256:26b54ddbbb9ee1d34d5d3668dd37d6cf74990ab23c828c2888dccdceee395594
colorama==0.4.3; sys_platform == "win32" \
    --hash=sha256:7d73d2a99753107a36ac6b455ee49046802e59d9d076ef8e47b61499fa29afff \
    --hash=sha256:e96da0d330793e2cb9485e9ddfd918d456036c7149416295932478192f4436a1
more-itertools==8.5.0 \
    --hash=sha256:6f83822ae94818eae2612063a5101a7311e68ae8002005b5e05f03fd74a86a20 \
    --hash=sha256:9b30f12df9393f0d28af9210ff8efe48d10c94f73e5daf886f10c4b0b0b4f03c
packaging==20.4 \
    --hash=sha256:998416ba6962ae7fbd6596850b80e17859a5753ba17c32284f67bfff33784181 \
    --hash=sha256:4357f74f47b9c12db93624a82154e9b120fa8293699949152b22065d556079f8
pluggy==0.13.1 \
    --hash=sha256:966c145cd83c96502c3c3868f50408687b38434af77734af1e9ca461a4081d2d \
    --hash=sha256:15b2acde666561e1298d71b523007ed7364de07029219b604cf808bfa1c765b0
py==1.9.0 \
    --hash=sha256:366389d1db726cd2fcfc79732e75410e5fe4d31db13692115529d34069a043c2 \
    --hash=sha256:9ca6883ce56b4e8da7e79ac18787889fa5206c79dcc67fb065376cd2fe03f342
pyparsing==2.4.7 \
    --hash=sha256:ef9d7589ef3c200abe66653d3f1ab1033c3c419ae9b9bdb1240a85b024efc88b \
    --hash=sha256:c203ec8783bf771a155b207279b9bccb8dea02d8f0c9e5f8ead507bc3246ecc1
pytest==5.4.3 \
    --hash=sha256:5c0db86b698e8f170ba4582a492248919255fcd4c79b1ee64ace34301fb589a1 \
    --hash=sha256:7979331bfcba207414f5e1263b5a0f8f521d0f457318836a7355531ed1a4c7d8
six==1.15.0 \
    --hash=sha256:8b74bedcbbbaca38ff6d7491d76f2b06b3592611af620f8426e82dddb04a5ced \
    --hash=sha256:30639c035cdb23534cd4aa2dd52c3bf48f06e5f4a941509c8bafd8ce11080259
wcwidth==0.2.5 \
    --hash=sha256:beb4802a9cebb9144e99086eff703a642a13d6a0052920003a230f3294bbe784 \
    --hash=sha256:c4d647b99872929fdb7bdcaa4fbe7f01413ed3d98077df798530e5b04f116c83
```

**Comment by cjolowicz** [comment]
Note that this makes the requirements.txt unusable with pip:

```
$ poetry export -f requirements.txt --dev | /venv/bin/pip install -r /dev/stdin
Collecting pytest==5.4.3
  Downloading pytest-5.4.3-py3-none-any.whl (248 kB)
     |████████████████████████████████| 248 kB 1.2 MB/s
Collecting attrs>=17.4.0
Collecting packaging
Collecting more-itertools>=4.0.0
Collecting pluggy<1.0,>=0.12
Collecting wcwidth
Collecting py>=1.5.0
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    attrs>=17.4.0 from https://files.pythonhosted.org/packages/14/df/479736ae1ef59842f512548bacefad1abed705e400212acba43f9b0fa556/attrs-20.2.0-py2.py3-none-any.whl#sha256=fce7fc47dfc976152e82d53ff92fa0407700c21acd20886a13777a0d20e655dc (from pytest==5.4.3->-r /dev/stdin (line 1))
    packaging from https://files.pythonhosted.org/packages/46/19/c5ab91b1b05cfe63cccd5cfc971db9214c6dd6ced54e33c30d5af1d2bc43/packaging-20.4-py2.py3-none-any.whl#sha256=998416ba6962ae7fbd6596850b80e17859a5753ba17c32284f67bfff33784181 (from pytest==5.4.3->-r /dev/stdin (line 1))
    more-itertools>=4.0.0 from https://files.pythonhosted.org/packages/ec/f7/9c33f38577d9fb9b1aa60c0fd9961d09574f5d47d2b830e5183e6adfc8cc/more_itertools-8.5.0-py3-none-any.whl#sha256=9b30f12df9393f0d28af9210ff8efe48d10c94f73e5daf886f10c4b0b0b4f03c (from pytest==5.4.3->-r /dev/stdin (line 1))
    pluggy<1.0,>=0.12 from https://files.pythonhosted.org/packages/a0/28/85c7aa31b80d150b772fbe4a229487bc6644da9ccb7e427dd8cc60cb8a62/pluggy-0.13.1-py2.py3-none-any.whl#sha256=966c145cd83c96502c3c3868f50408687b38434af77734af1e9ca461a4081d2d (from pytest==5.4.3->-r /dev/stdin (line 1))
    wcwidth from https://files.pythonhosted.org/packages/59/7c/e39aca596badaf1b78e8f547c807b04dae603a433d3e7a7e04d67f2ef3e5/wcwidth-0.2.5-py2.py3-none-any.whl#sha256=beb4802a9cebb9144e99086eff703a642a13d6a0052920003a230f3294bbe784 (from pytest==5.4.3->-r /dev/stdin (line 1))
    py>=1.5.0 from https://files.pythonhosted.org/packages/68/0f/41a43535b52a81e4f29e420a151032d26f08b62206840c48d14b70e53376/py-1.9.0-py2.py3-none-any.whl#sha256=366389d1db726cd2fcfc79732e75410e5fe4d31db13692115529d34069a043c2 (from pytest==5.4.3->-r /dev/stdin (line 1))
WARNING: You are using pip version 20.2.2; however, version 20.2.3 is available.
You should consider upgrading via the '/venv/bin/python -m pip install --upgrade pip' command.
```

**Comment by abn** [comment]
This was resolved with #3024.

**Comment by thedrow** [comment]
@abn This was incorrectly marked as `To do` in both boards. This is done :)
